src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/RankSumTest.java

```diff
@@ -47,16 +50,32 @@ public RankSumTest(){
         final int refLoc = vc.getStart();
 
         if( likelihoods != null) {
-            for (final ReadLikelihoods<Allele>.BestAllele bestAllele : likelihoods.bestAlleles()) {
-                final GATKRead read = bestAllele.read;
-                final Allele allele = bestAllele.allele;
-                if (bestAllele.isInformative() && isUsableRead(read, refLoc)) {
-                    final OptionalDouble value = getElementForRead(read, refLoc, bestAllele);
-                    // Bypass read if the clipping goal is not reached or the refloc is inside a spanning deletion
-                    if (value.isPresent() && value.getAsDouble() != INVALID_ELEMENT_FROM_READ) {
-                        if (allele.isReference()) {
+            // Default to using the likelihoods to calculate the rank sum
+            if (likelihoods.hasFilledLiklihoods()) {
+                for (final ReadLikelihoods<Allele>.BestAllele bestAllele : likelihoods.bestAlleles()) {
+                    final GATKRead read = bestAllele.read;
+                    final Allele allele = bestAllele.allele;
+                    if (bestAllele.isInformative() && isUsableRead(read, refLoc)) {
+                        final OptionalDouble value = getElementForRead(read, refLoc, bestAllele);
+                        // Bypass read if the clipping goal is not reached or the refloc is inside a spanning deletion
+                        if (value.isPresent() && value.getAsDouble() != INVALID_ELEMENT_FROM_READ) {
+                            if (allele.isReference()) {
+                                refQuals.add(value.getAsDouble());
+                            } else if (vc.hasAllele(allele)) {
+                                altQuals.add(value.getAsDouble());
+                            }
+                        }
+                    }
+                }
+
+            // Use the pileup to stratify otherwise
+            } else if (!(this instanceof AS_RankSumTest)) {
+                for (final PileupElement p : likelihoods.getStratifiedPileups(vc).values().stream().flatMap(Collection::stream).collect(Collectors.toList())) {
+                    final OptionalDouble value = getElementForPileupElement(p, refLoc);
+                    if (value.isPresent() && value.getAsDouble() != INVALID_ELEMENT_FROM_READ && isUsableBase(p)) {
+                        if (vc.getReference().equals(Allele.create(p.getBase(), true))) {
```

Comment:
It will return the first base before the deletion site. Which is not ideal I imagine but this seems to be a repeated bug from gatk3. It seems like an edge case because the only time this could make a difference is if you are at a site that is not marked as a deletion but you then have a read which has a deletion on it and then happens to match one of the two alleles of the snp... So a minor effect. Should I open a new ticket for this?